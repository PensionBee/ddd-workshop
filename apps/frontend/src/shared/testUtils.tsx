import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
