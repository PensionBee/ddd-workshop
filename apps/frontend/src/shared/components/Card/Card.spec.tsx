import { render } from "@shared/testUtils";
import { vi } from "vitest";
import {
  Card,
  _CardBaseComponent,
  _CardButtonComponent,
  _CardLinkComponent,
} from "./Card";

describe("Card", () => {
  describe("Default", () => {
    it("should render correctly", () => {
      const { asFragment } = render(
        <Card>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Card>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});

// Make sure story components are not rendered
describe("Card.stories", () => {
  it("_CardBaseComponent should return null", () => {
    expect(_CardBaseComponent({ children: null })).toBe(null);
  });
  it("_CardButtonComponent should return null", () => {
    expect(_CardButtonComponent({ as: "button" })).toBe(null);
  });
  it("_CardLinkComponent should return null", () => {
    expect(_CardLinkComponent({ as: "a" })).toBe(null);
  });
});
