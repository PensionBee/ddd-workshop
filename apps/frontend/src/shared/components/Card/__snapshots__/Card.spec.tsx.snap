// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Default should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe iQfGuE"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </div>
</DocumentFragment>
`;
